packages/opentelemetry-sdk-trace-base/src/Span.ts

```diff
@@ -235,4 +235,62 @@ export class Span implements api.Span, ReadableSpan {
     }
     return this._ended;
   }
+
+  // Utility function to truncate given value within size
+  // for value type of string, will truncate to given limit
+  // for type of non-string, will return same value
+  private _truncateToLimitUtil(value: string, limit: number): string {
+    if (typeof value != 'string' || value.length <= limit) {
+      return value;
+    }
+    return value.substr(0, limit);
+  }
+
+  // Check whether given value is array of strings or not
+  private _isArrayOfStrings(value: unknown): boolean {
+    if (!Array.isArray(value)) {
+      return false
+    }
+    return value.every(val => typeof val == 'string')
+  }
+
+  /**
+   * If the given attribute value is of type string and has more characters than given {@code attributeValueLengthLimit} then
+   * return string with trucated to {@code attributeValueLengthLimit} characters
+   *
+   * If the given attribute value is array of strings then
+   * return new array of strings with each element truncated to {@code attributeValueLengthLimit} characters
+   *
+   * Otherwise return same Attribute {@code value}
+   *
+   * @param value Attribute value
+   * @returns truncated attribute value if required, otherwise same value
+   */
+  private _truncateToSize(value?: SpanAttributeValue): SpanAttributeValue | undefined {
+    const limit = this._spanLimits.attributeValueLengthLimit!
+    // Check limit
+    if (typeof limit != 'number' || limit <= 0) {
```

Comment:
The types claim the limit can be undefined which would trigger this if-condition

Reply:
As per [spec](https://github.com/open-telemetry/opentelemetry-specification/blob/c5f1a70340059953244fcc9aabb371347e0989fc/specification/sdk-environment-variables.md#attribute-limits),  Says `Empty value is treated as infinity. Non-integer and negative values are invalid.` 

when limit is not set we will get limit as `undefined` type.
So we need to skip truncating attribute value for these cases.

This if condition will handle that.

Let me know if that make sense.

Reply:
ok

Reply:
Can you add a comment along the lines of "undefined limit means do not truncate"

Reply:
I added comment as you mentioned